src/frontend/src/layouts/PageBlock.tsx

```diff
@@ -46,19 +46,28 @@ const PageBlock: React.FC<PageBlockProps> = ({ title, headerRight, children, sty
         }
       >
         {title && (
-          <Box sx={{ display: 'flex', flexDirection: 'row', mb: 1 }}>
-            <Typography variant="h5" sx={{ flexGrow: 1 }}>
-              {title}
+          <Box sx={{ mb: 1 }}>
+            <Typography variant="h5" sx={{ display: 'flex', justifyContent: 'space-between' }}>
+              <Typography
+                variant="inherit"
+                onClick={() => {
+                  setCollapsed(!collapsed);
+                }}
+              >
+                {collapsed ? (
+                  <ExpandMoreIcon sx={{ ml: -1, paddingRight: 0.5 }} />
+                ) : (
+                  <ExpandLessIcon sx={{ ml: -1, paddingRight: 0.5 }} />
+                )}
+                {title}
+              </Typography>
+              {headerRight}
             </Typography>
```

Comment:
you should only need 1 box and 1 typography here, the typography only needs to wrap the icon and the title, and the box should wrap the typography and the headerRight. You should be able to shuffle the props around and it'll still look like how it does rn